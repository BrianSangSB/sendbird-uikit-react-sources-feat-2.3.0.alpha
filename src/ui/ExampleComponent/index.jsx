import React from 'react';
import PropTypes from 'prop-types';

import './index.scss';

export default function ExampleComponent({ text }) {
  return (
    <div>
      This is an example :
      <span className="sendbird-text">
        {text}
      </span>
    </div>
  );
}

ExampleComponent.propTypes = {
  text: PropTypes.string.isRequired,
};
