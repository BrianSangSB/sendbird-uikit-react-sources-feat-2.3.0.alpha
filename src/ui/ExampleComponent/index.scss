@import '../../styles/variables';

.sendbird-text {
  @include themed() {
    color: t(primary-1);
  }
}
