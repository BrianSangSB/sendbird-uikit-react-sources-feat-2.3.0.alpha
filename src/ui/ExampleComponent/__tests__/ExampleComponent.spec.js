import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

import ExampleComponent from "../index";

describe('ExampleComponent', () => {
  it('should render text prop', function() {
    const text = "example-text";
    const component = shallow(<ExampleComponent text={text} />);

    expect(
      component.find(".sendbird-text").text()
    ).toEqual(text);
  });

  it('should do a snapshot test of the ExampleComponent DOM', function() {
    const text = "example-text";
    const component = renderer.create(
      <ExampleComponent text={text} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
