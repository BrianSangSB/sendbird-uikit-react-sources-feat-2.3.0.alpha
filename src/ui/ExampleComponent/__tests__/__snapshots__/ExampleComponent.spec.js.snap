// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExampleComponent should do a snapshot test of the ExampleComponent DOM 1`] = `
<div>
  This is an example :
  <span
    className="sendbird-text"
  >
    example-text
  </span>
</div>
`;
